### ISSUE TITLE
Positional parameter doesn't handle validation error

### ISSUE THREAD

**Issue by Ravencentric** [BrianPugh/cyclopts#25]
Take this cli app for example:
```py
from pathlib import Path
from typing import Annotated

from cyclopts import App, Parameter, validators

app = App()

@app.default() # type: ignore
def cli(
    path: Annotated[
        Path, 
        Parameter(
            help="directory or file",
            validator=validators.Path(exists=True))
        ] = Path.cwd(),
    /,
    *,
    config: Annotated[
        Path, 
        Parameter(
            name=["--config"], 
            help="config file", 
            validator=validators.Path(exists=True))
        ] = Path.cwd() / "config.yaml",
) -> None:
    print(path)
    print(config)
    exit()

if __name__ == "__main__":
    app()
```
If I pass an invalid path to `--config`, it handles it fine
```py
> python .\test.py --config "doesntexist.yml"
╭─ Error ─────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Invalid value for --config. doesntexist.yml does not exist.                                                     │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```
However, if i pass an invalid path to the positional parameter `path` then:
```py
> python .\test.py "doesntexist"
Traceback (most recent call last):
  File "C:\Users\raven\Documents\GitHub\test\.venv\Lib\site-packages\cyclopts\bind.py", line 314, in _convert
    cparam.validator(type_, val)
  File "C:\Users\raven\Documents\GitHub\test\.venv\Lib\site-packages\cyclopts\validators\_path.py", line 33, in __call__
    raise ValueError(f"{path} does not exist.")
ValueError: doesntexist does not exist.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\raven\Documents\GitHub\test\.venv\Lib\site-packages\cyclopts\core.py", line 411, in parse_args
    command, bound, unused_tokens = self.parse_known_args(tokens, console=console)
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\test\.venv\Lib\site-packages\cyclopts\core.py", line 328, in parse_known_args
    bound, unused_tokens = create_bound_arguments(command, unused_tokens)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\test\.venv\Lib\site-packages\cyclopts\bind.py", line 357, in create_bound_arguments
    coerced = _convert(mapping)
              ^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\test\.venv\Lib\site-packages\cyclopts\bind.py", line 321, in _convert
    raise new_exception from e
cyclopts.exceptions.ValidationError: <exception str() failed>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\raven\Documents\GitHub\test\test.py", line 30, in <module>
    app()
  File "C:\Users\raven\Documents\GitHub\test\.venv\Lib\site-packages\cyclopts\core.py", line 453, in __call__
    command, bound = self.parse_args(tokens, **kwargs)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\test\.venv\Lib\site-packages\cyclopts\core.py", line 423, in parse_args
    console.print(format_cyclopts_error(e))
                  ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\test\.venv\Lib\site-packages\cyclopts\exceptions.py", line 261, in format_cyclopts_error
    Text(str(e), "default"),
         ^^^^^^
  File "C:\Users\raven\Documents\GitHub\test\.venv\Lib\site-packages\cyclopts\exceptions.py", line 143, in __str__
    parameter_cli_name = ",".join(self.parameter2cli[self.parameter])
                                  ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
KeyError: <Parameter "path: Annotated[pathlib.Path, Parameter(name=(), converter=None, validator=Path(exists=True, file_okay=True, dir_okay=True), negative=None, token_count=None, parse=True, show=None, show_default=None, show_choices=None, help='directory or file', show_env_var=None, env_var=())] = WindowsPath('C:/Users/raven/Documents/GitHub/test')">
```

**Comment by BrianPugh** [comment]
thanks for the bug report! fixing now.

**Comment by BrianPugh** [comment]
this should be fixed in the newly release v1.1.1. Please let me know if this fixed it for you!

**Comment by Ravencentric** [comment]
Yes!
```
> python .\test.py "fakepath"
╭─ Error ──────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Invalid value for PATH. fakepath does not exist.                                                                     │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```
